src/mutations/creators.ts

```diff
@@ -53,24 +54,46 @@ export const createTypeAdditionMutation = (
 		) ||
 		isKnownGlobalBaseType(declaredType)
 	) {
-		return {
-			insertion: ` ${newTypeAlias}`,
-			range: {
-				begin: node.type.pos,
-				end: node.type.end,
-			},
-			type: "text-swap",
-		};
+		return textSwap(` ${newTypeAlias}`, node.type.pos, node.type.end);
+	}
+
+	const isAsyncFunction =
+		ts.isFunctionDeclaration(node) &&
+		node.modifiers?.some((modifier) => tsutils.isAsyncKeyword(modifier));
+
+	if (!isAsyncFunction) {
+		// Create a mutation insertion that adds the missing types in
+		return textInsert(` | ${newTypeAlias}`, node.type.end);
+	}
+
+	const typeChecker = request.services.languageService
+		.getProgram()
+		?.getTypeChecker();
```

Comment:
[Types] The type checker should never be undefined. One is available under:

```suggestion
	const typeChecker = request.services.program.getTypeChecker();
```